src/Recordings.cpp

```diff
@@ -296,58 +335,66 @@ bool Recordings::UpdatePvrRecording(TiXmlElement* pRecordingNode, kodi::addon::P
   return true;
 }
 
-bool Recordings::GetAdditiveString(const TiXmlNode* pRootNode, const char* strTag, const std::string& strSeparator, std::string& strStringValue, bool clear)
+bool Recordings::ParseNextPVRSubtitle(TiXmlElement *pRecordingNode, kodi::addon::PVRRecording& tag)
 {
-  bool bResult = false;
-  if (pRootNode != nullptr)
+  std::string buffer;
+  bool hasSeasonEpisode = false;
+  if (XMLUtils::GetString(pRecordingNode, "subtitle", buffer))
   {
-    std::string strTemp;
-    const TiXmlElement* node = pRootNode->FirstChildElement(strTag);
-    if (node && node->FirstChild() && clear)
-      strStringValue.clear();
-    while (node)
+    std::regex base_regex("S(\\d\\d)E(\\d+) - ?(.+)?");
+    std::smatch base_match;
+    // note NextPVR does not support S0 for specials
+    if (std::regex_match(buffer, base_match, base_regex))
     {
-      if (node->FirstChild())
+      if (base_match.size() == 3 || base_match.size() == 4)
       {
-        bResult = true;
-        strTemp = node->FirstChild()->Value();
-        const char* clear = node->Attribute("clear");
-        if (strStringValue.empty() || (clear && StringUtils::CompareNoCase(clear, "true") == 0))
-          strStringValue = strTemp;
-        else
-          strStringValue += strSeparator + strTemp;
+
+        std::ssub_match base_sub_match = base_match[1];
+        tag.SetSeriesNumber(std::stoi(base_sub_match.str()));
+        base_sub_match = base_match[2];
+        tag.SetEpisodeNumber(std::stoi(base_sub_match.str()));
```

Comment:
About `std::stoi` I'm not sure, Maybe the `atoi`?
Heard from @phunkyfish that this is not safe. I'm right?

Reply:
Stoi won’t return 0 if it can’t read a numeric value. So you may want around instead.

Reply:
> 
> 
> Stoi won’t return 0 if it can’t read a numeric value. So you may want atoi instead.

Ah I see now I would need try/catch  in some cases.   Most of them are safe, including this one because they follow regex \d but this is important to know.  Thanks both of you.